controller/volume_controller.go

```diff
@@ -1028,6 +1043,10 @@ func (vc *VolumeController) ReconcileVolumeState(v *longhorn.Volume, es map[stri
 			vc.eventRecorder.Eventf(v, v1.EventTypeNormal, EventReasonDetached, "volume %v has been detached", v.Name)
 		}
 		// Automatic reattach the volume if PendingNodeID was set
+		// TODO: need to revisit for CurrentNodeID and PendingNodeID update
+		// The reason is usually when CurrentNodeID is set, the volume
+		// should be in attaching or attached state, currently
+		// CurrentNodeID is set when volume is in detached state.
```

Comment:
Add one more statement:
And we will implicitly assume `v.status.CurrentNodeID` has been the final state of this sync loop if this code block is reached. Hence updating `v.status.CurrentNodeID` here is weird.

Reply:
Will do. Thanks.